File: Project/dsa/python/1367-linked-list-in-binary-tree/solution.py
# 1367. Linked List in Binary Tree  [Medium]
# https://leetcode.com/problems/linked-list-in-binary-tree/
#
# Approach:
#
# Complexity: Time O(?)  Space O(?)

from typing import *

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSubPath(self, head: ListNode | None, root: TreeNode | None) -> bool:
        pass


if __name__ == "__main__":
    s = Solution()
    # print(s.method(...))

File: Project/dsa/java/1367-linked-list-in-binary-tree/Solution.java
// 1367. Linked List in Binary Tree  [Medium]
// https://leetcode.com/problems/linked-list-in-binary-tree/
//
// Approach:
//
// Complexity: Time O(?)  Space O(?)

import java.util.*;

class Solution {
    public boolean isSubPath(ListNode head, TreeNode root) {
        
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        // System.out.println(s.method(...));
    }
}

// Definition for singly-linked list.
class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

// Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
